### ISSUE TITLE
WSL2 Ubuntu ARM64 build error

### ISSUE THREAD

**Issue by pgovind** [dotnet/runtime#38958]
Just logging an issue I ran into while building dotnet/runtime on WSL2 Arm64:

```
_$ ./build.sh clr -c release_
.
.
.
Build FAILED.

/home/prgovi/repos/runtime_2/src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj(87,5): error MSB3030: Could not copy the file "/home/prgovi/repos/runtime_2/artifacts/bin/coreclr/Linux.arm64.Release//x64/libjitinterface.so" because it was not found.
/home/prgovi/.nuget/packages/microsoft.build.traversal/2.0.34/Sdk/Sdk.targets(116,5): error MSB4181: The "MSBuild" task returned false but did not log an error. [/home/prgovi/repos/runtime_2/Build.proj]
    0 Warning(s)
    2 Error(s)
```

I'm not super familiar with the build infra, but it looks like a bad path? I see a **"/x64"** in the error message. Running `find . -name libjitinterface*` gives this:
```
./artifacts/obj/coreclr/Linux.arm64.Release/src/tools/aot/jitinterface/libjitinterface.so
./artifacts/bin/coreclr/Linux.arm64.Release/libjitinterface.so
./artifacts/bin/coreclr/Linux.arm64.Release/crossgen2/libjitinterface.so
```

@carlossanlop

**Comment by safern** [comment]
cc: @jkoritzinsky @jashook @trylek

**Comment by carlossanlop** [comment]
I hit this too:
```
/home/carlos/runtime/src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj(87,5):
error MSB3030:
Could not copy the file "/home/carlos/runtime/artifacts/bin/coreclr/Linux.arm64.Release//x64/libjitinterface.so" because it was not found.
```

I think the problem is in the path - notice the second string has a double `//` - I suspect the string was expecting something to get embedded in there.

**Comment by safern** [comment]
The problem is here:

https://github.com/dotnet/runtime/blob/3527fdefde316439b0b28f88febd02fcd93ea34b/src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj#L83

since that is using `CrossHostArch` which defaults to `x64` if `TargetArchitecture == arm64` -- I think this is very specialized for cross build inside an x64 host. 

https://github.com/dotnet/runtime/blob/3527fdefde316439b0b28f88febd02fcd93ea34b/src/coreclr/src/tools/aot/crossgen2/crossgen2.csproj#L45

As a temporary workaround you could just pass that property globally as empty: `/p:CrossHostArch=`. 

@trylek @jkoritzinsky maybe we should check the host architecture instead to see if it differs from `TargetArchitecture` or is there a way to know if it is a cross build?

**Comment by jkoritzinsky** [comment]
Yes we should. We have checks in the installer build for not using the "cross host arch" when Host = Target Arch.

**Comment by safern** [comment]
Yeah, `$(BuildArchitecture)` is the property. 

https://github.com/dotnet/runtime/blob/74b173787bad85eac022b9fb680f247aeda76305/Directory.Build.props#L103

And we have some properties for this apparently:
https://github.com/dotnet/runtime/blob/74b173787bad85eac022b9fb680f247aeda76305/eng/liveBuilds.targets#L77

We should probably just use that property instead to construct the file path.

**Comment by AndyAyersMS** [comment]
Hitting this now too, building natively for arm64 on a Macbook.
